sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.AotCompatibilityTestApp/Azure.Monitor.OpenTelemetry.Exporter.AotCompatibilityTestApp.csproj

```diff
@@ -0,0 +1,34 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFrameworks>net7.0;net6.0;net462</TargetFrameworks>
+    <IsTestSupportProject>true</IsTestSupportProject>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(TargetFramework)' == 'net7.0'">
+    <PublishAot>true</PublishAot>
+    <TrimmerSingleWarn>false</TrimmerSingleWarn>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\..\..\..\core\Azure.Core\src\Azure.Core.csproj" />
+    <ProjectReference Include="..\..\src\Azure.Monitor.OpenTelemetry.Exporter.csproj" />
+
+    <!-- Update this dependency to its latest, which has all the annotations -->
+    <PackageReference Include="Microsoft.Extensions.Logging.Configuration" VersionOverride="8.0.0-rc.1.23419.4" />
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(TargetFramework)' == 'net7.0'">
```

Comment:
Nit - change this to `net6.0` - one day it would be nice to also run this app targeting `net6.0` and `PublishTrimmed=true` to also validate pure trim compatibility (it's using different tooling in the SDK than AOT, so potentially slightly different behavior - although we try hard to keep them in sync).

Reply:
Hmm. I've been doing all my testing with net7. 

https://github.com/Azure/azure-sdk-for-net/blob/2532c7e8fe589201fe7907ce18e58a4a19bc28fa/sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.AotCompatibilityTestApp/Azure.Monitor.OpenTelemetry.Exporter.AotCompatibilityTestApp.csproj#L8-L11

https://github.com/Azure/azure-sdk-for-net/blob/2532c7e8fe589201fe7907ce18e58a4a19bc28fa/sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.AotCompatibilityTestApp/Azure.Monitor.OpenTelemetry.Exporter.AotCompatibilityTestApp.csproj#L21-L23

https://github.com/Azure/azure-sdk-for-net/blob/2532c7e8fe589201fe7907ce18e58a4a19bc28fa/sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.AotCompatibilityTestApp/test-aot-compat.ps1#L4

@vitek-karas Should this work with net6? Testing net6 locally doesn't work for me.

Reply:
This was just "looking forward". AOT is only supported on .NET 7+. But trimming alone is supported on .NET 6+. It would be nice to be able to test that just trimming works on 6. You would have to not set `PublishAot` and instead set `PublishTrimmed=true`. Don't worry about it... I'm just nit picking... sorry.

Reply:
No sorry necessary. I'm experimenting and learning. Thank you for your help here!